### PR TITLE
Request cache

### PR DESCRIPTION
**Pull request by h2kuo**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @h2kuo has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Cache/BOXRequestCache.m [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        } else if (!isDirectory) {
+            BOXLog(@"Unable to create BOXRequestCache directory at %@ because a file already exists at that location.", _cachePath);
+            return nil;
+        }
+        
+        _cache = [[PINCache alloc] initWithName:@"BoxRequestCache" rootPath:self.cachePath];
+        
+        // This will allow the cache to be automatically be trimmed whenever it exceeds the byte limit.
+        // In the future we can expose an interface to customize this.
+        _cache.diskCache.byteLimit = 500*1024*1024;
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Cache/BOXRequestCache.m:75]: Let's do something a bit smaller. I suggest 100MB

File: BoxContentSDK/BoxContentSDK/Categories/NSString+BOXAdditions.m [hunk 2]
```diff
@@ -60,4 +61,22 @@ - (BOOL)box_isEmptyOrWhitespacesOnly
     return isEmptyOrWhitespaces;
 }
 
+- (NSString *)box_sha1
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Categories/NSString+BOXAdditions.m:64]: Did you copy this from somewhere? If so, please add a comment to give credit.

File: BoxContentSDK/BoxContentSDK/Cache/BOXRequestCache.m [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+@implementation BOXRequestCache
+
+- (instancetype)initWithUserID:(NSString *)userID
+{
+    return [self initWithUserID:userID cacheDirectory:nil];
+}
+
+- (instancetype)initWithUserID:(NSString *)userID cacheDirectory:(NSURL *)cacheDirectory
+{
+    self = [super init];
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Cache/BOXRequestCache.m:35]: self = [self initWithUserID:userID]

File: BoxContentSDK/BoxContentSDK/Cache/BOXRequestCache.m [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+@property (nonatomic, readwrite, copy) NSString *userID;
+@property (nonatomic, readwrite, strong) NSURL *cacheDirectory;
+
+@end
+
+@implementation BOXRequestCache
+
+- (instancetype)initWithUserID:(NSString *)userID
+{
+    return [self initWithUserID:userID cacheDirectory:nil];
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Cache/BOXRequestCache.m:30]: self = [super init];
_userID = userID

File: BoxContentSDK/BoxContentSDK/Cache/BOXRequestCache.m [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+- (instancetype)initWithUserID:(NSString *)userID cacheDirectory:(NSURL *)cacheDirectory
+{
+    self = [super init];
+    
+    if (self) {
+        _userID = userID;
+        
+        if (!cacheDirectory) {
+            cacheDirectory = [NSURL URLWithString:[NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES) objectAtIndex:0]];
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Cache/BOXRequestCache.m:41]: Do this in initWithUserID:

**Pull request closed without merging**